vm/src/types/slot.rs

```diff
@@ -469,6 +481,194 @@ impl PyType {
                 toggle_ext_func!(number_methods, float, float_wrapper);
                 update_pointer_slot!(as_number, number_methods);
             }
+            _ if name == identifier!(ctx, __add__) => {
+                toggle_ext_func!(number_methods, add, number_binary_op_wrapper!(__add__));
+                update_pointer_slot!(as_number, number_methods);
+            }
```

Comment:
And then this part will look like:
```suggestion
            _ if name == identifier!(ctx, __add__)  || name == identifier!(ctx, __radd__)=> {
                toggle_ext_func!(number_methods, add, number_binary_op_wrapper!(__add__, __radd__)); // no toggle but ok for a while
                update_pointer_slot!(as_number, number_methods);
            }
```